src/main/java/org/sonatype/goodies/packageurl/jaxrs/PackageUrlParamConverter.java

```diff
@@ -43,13 +42,19 @@ public PackageUrlParamConverter(@Nullable final RenderFlavor flavor) {
 
   @Override
   public PackageUrl fromString(final String value) {
-    checkArgument(value != null);
+    // Do not use requireNonNull(value); since the interface requires an IllegalArgumentException()
```

Comment:
What if instead of handling this specially and adding an inline comment we have our own checkArgument() that throws an IllegalArgumentException? Then we can use it everywhere in this PR and know we didn't change anything from the Guava way.

We could also get rid of these inline comments in that case.

Reply:
🤷‍♂ It's just two places in the same file. Didn't seem work it. Again LMK...

Reply:
I agree with @andrewstein here, and its the same thing I did in my PR to remove guava, since its only this one jaxrs component that explicitly needs IAE.  But if there were a lot more then I would agree with @bradcupit.

As for the comments, well Its actually on the javadoc of the interface, but the added comment helps folks that wouldn't otherwise go look at the interface docs they are implementing... /me shrugs :-P

Reply:
my primary motivation was to get rid of the inline comment, which could be moved to our own checkArgument() method.

If you guys are fine with it then I'm not going to push for it.